### ISSUE TITLE
bug: Localstack v3+ breaks in-cluster DNS resolution

### ISSUE THREAD

**Issue by WarpRat** [localstack/localstack#9955]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running our development environments in kubernetes our lambdas call services by name relying on kube-dns to resolve the svc to the running pod(s). After updating to localstack v3+ this is no longer working. [This PR](https://github.com/localstack/localstack/pull/9444) changed the way localstack adds its own nameserver to the localstack container's `/etc/resolv.conf`. In previous versions the localstack entry was prepended to the existing `/etc/resolv.conf` but now overwrites the whole file, which drops the existing injected kube-dns settings. If I manually edit the container `resolve.conf` to include the dropped kube-dns portion, everything works as expected.

Localstack is properly detecting the kube-dns address as the fallback DNS server but I believe it doesn't work as expected because it's missing the search domains and ndots options. If there's a reason localstack can't go back to injecting an entry into the conf file, an option to automatically enable in-cluster resolution would be very helpful. 

The following `/etc/resolv.conf` works as expected:

```
# The following line is required by LocalStack
nameserver 127.0.0.1


nameserver 164.254.20.10
search localstack-dns-test.svc.cluster.local svc.cluster.local cluster.local us-west-2.compute.internal
options ndots:5
```

### Expected Behavior

I expect lambdas running in the localstack dind sidecar to be able to resolve kubernetes services using standard in-cluster DNS (kube-dns/core-dns)

### How are you starting LocalStack?

Custom:

with a helm chart in kubernetes, both Rancher and minikube

### Steps To Reproduce

Deploy localstack with the latest helm chart on v3+ and attempt to call any kubernetes service from inside a lambda container (or anywhere else from inside the localstack or dind containers). 

### Environment

```markdown
- OS: Debian bookworm (localstack container). Ubuntu 24.04 (node)
- LocalStack: >= v3
```

**Comment by dfangl** [comment]
Hi @WarpRat !
I can see your current issue with the implementation. 
We started completely overriding the file due to issues where certain requests were not hitting the LocalStack DNS, but the other configured one (due to possible timeouts etc, but it should retry the LS one in those cases).

For now, you can always use `DNS_ADDRESS=0` to completely disable LocalStack's DNS logic. This will however lead to some DNS names not pointing to LocalStack as expected anymore (like `localhost.localstack.cloud`), and transparent endpoint injection in pro will cease to function, but it might be a viable workaround, depending on your scenario.

To fix the underlying issue, would it suffice for you if LocalStack continues to be the only nameserver listed, but the search domains and options will be retained?

**Comment by WarpRat** [comment]
Thanks for the response @dfangl

Yeah, the `DNS_ADDRESS=0` setting should work just fine for us, I don't know how I missed that previously. Presumably that doesn't have any impact on localstack services themselves being able to communicate with other localstack services, it would only affect anything that's not local to the pod that doesn't have the proper `AWS_ENDPOINT_URL` value for the localstack service.

It had been a while since I looked at this but I just spun up a test pod in our infra and it does look like localstack properly identifies the special node-local-dns endpoint as the upstream:
```
2024-03-01T22:30:34.426 DEBUG --- [  MainThread] localstack.dns.server      : Determined fallback dns: 164.254.20.10
```
I think keeping the search domains and options would work for almost all kube configurations.

**Comment by dfangl** [comment]
This issue should be fixed with #10731, we now correctly preserve the search domains and options, and just replace the nameserver in the resolv.conf file.

Can you confirm the DNS resolution in the cluster now works as expected?
